pkg/controllers/osc/resources/operating_system_config.go

```diff
@@ -97,22 +85,23 @@ func OperatingSystemConfigCreator(
 			}
 
 			data := filesData{
-				KubeletVersion:       md.Spec.Template.Spec.Versions.Kubelet,
-				KubeletConfiguration: kubeletConfig,
-				KubeletSystemdUnit:   kubeletSystemdUnit,
-				CNIVersion:           cniVersion,
-				ClusterDNSIPs:        clusterDNSIPs,
-				KubernetesCACert:     CACert,
-				Kubeconfig:           kubeconfigStr,
-				ContainerRuntime:     "docker",
-				CloudProviderName:    cloudProvider.Name,
-				Hostname:             "Node-1", // FIX this shit
-				ExtraKubeletFlags:    KubeletFlags(),
-
+				KubeletVersion:             md.Spec.Template.Spec.Versions.Kubelet,
+				CNIVersion:                 cniVersion,
+				ClusterDNSIPs:              clusterDNSIPs,
+				KubernetesCACert:           CACert,
+				Kubeconfig:                 kubeconfigStr,
+				ContainerRuntime:           containerRuntime,
+				CloudProviderName:          cloudProvider.Name,
+				Hostname:                   "Node-1", // FIX this shit
```

Comment:
Getting rid of this is quite problematic: in MC the hostname is derived from the machine name, but the controller reconciles the `MachineDeployment`. Hence, we could either consider the tight coupling between the node name and the machine name or reconcile the `Machines` instead of the `MachineDeployments`.